api/histograms.graphql

```diff
@@ -1,113 +1,113 @@
-type Histogram {
-  """
-  - `id`: unique integer identifier of a histogram
-  - `name`: String, can be duplicate
-  - `data`: List of x y coordinates
-  - `xrange`: Range of the x axis to plot
-  - `yrange`: Range of the y axis to plot
-  - `len`: Number of data points
-  - `type`: Additional meta data
-  - `created`: Datetime when histogram was made in the database
-  """
-  id: ID!
-  name: String
-  data: [Point!]
-  xrange: Range!
-  yrange: Range!
-  len: Int
-  type: String
-  created: Datetime
-}
-
-type LiveHistogram {
-  """
-  - `id`: unique integer identifier of a histogram
-  - `name`: String, can be duplicate
-  - `data`: List of x y coordinates
-  - `xrange`: Range of the x axis to plot
-  - `yrange`: Range of the y axis to plot
-  - `len`: Number of data points
-  - `current`: The last added `Point`
-  - `type`: Additional meta data
-  - `created`: Datetime when histogram was made in the database
-  """
-  id: ID!
-  name: String
-  data: [Point!]
-  xrange: Range!
-  yrange: Range!
-  len: Int
-  current: Point
-  type: String
-  created: Datetime
-}
-
-input HistogramInput {
-  """
-  - `id`: unique integer identifier of a histogram
-  - `name`: String, can be duplicate
-  - `data`: List of x y coordinates
-  - `xrange`: Range of the x axis to plot
-  - `yrange`: Range of the y axis to plot
-  - `type`: Additional meta data
-  - `isLive`: Whether histogram gets put in the live database or the static one (default: static)
-  """
-  id: ID!
-  name: String
-  data: [PointInput!]
-  xrange: RangeInput!
-  yrange: RangeInput!
-  type: String
-  isLive: Boolean
-}
-
-input HistogramUpdateInput {
-  """
-  - `id`: unique integer identifier of a histogram
-  - `name`: String, can be duplicate
-  - `data`: List of x y coordinates
-  - `xrange`: Range of the x axis to plot
-  - `yrange`: Range of the y axis to plot
-  - `type`: Additional meta data
-  - `isLive`: Whether histogram gets put in the live database or the static one (default: static)
-  """
-  id: ID!
-  name: String
-  data: [PointInput!]
-  xrange: RangeInput
-  yrange: RangeInput
-  type: String
-  isLive: Boolean
-}
-
-type Point {
-  "coordinate in a format compatible with an SVG plotter"
-  x: Float!
-  y: Float!
-}
-
-input PointInput {
-  "coordinate in a format compatible with an SVG plotter"
-  x: Float!
-  y: Float!
-}
-
-type Range {
-  "Gives the minimum and maximum range of an axis"
-  min: Float!
-  max: Float!
-}
-
-input RangeInput {
-  "Gives the minimum and maximum range of an axis"
-  min: Float!
-  max: Float!
-}
-
-type HistogramPayload {
-  message: String!
-  success: Boolean!
-}
-
-"Datetime is in UTC"
-scalar Datetime
+type Histogram {
```

Comment:
What is the motivation for naming all the directory for the graphql files `api`?

I think renaming the directory to `schema` or something would make more sense. Alternatively, putting the graphql files within their respective apps in the `app` directory would work too

There aren't a lot of django ariadne examples out there, but here's one for reference:
https://stackoverflow.com/questions/59509069/folder-design-when-using-ariadne-in-django

Reply:
Yeah my old organization method was to have the graphql files in apps, but since ariadne does not allow you to comfortably divide up your queries/mutations into separate files I still had a large schema file located in the main app directory

My primary reason for putting all the schema files in one folder is because when I changed things in one schema file I was very likely going to edit a second schema file.

I've seen a few ariadne django examples when I was learning it, and in most of those it makes sense to put the schema files with the app because they use django for both FE and BE. So for instance a page that displays histograms will be located in the histogram_app directory, along with the schema, resolvers, models, etc. 

However, for our implementation, we have a FE and ariadne is very minimally coupled with the rest of the django framework. I also found that example projects with all the API grouped clearly in one place was much easier to parse than one with schema files scattered all over the place.